### ISSUE TITLE
Aptly disrespects umask

### ISSUE THREAD

**Issue by Hawk777** [aptly-dev/aptly#1200]
Aptly creates files and directories in `db` and `pool` which do not respect my umask.

## Detailed Description
```
$ cat .aptly.conf 
{
  "rootDir": "/home/me/aptly"
}
$ umask
0002
$ aptly_1.5.0_linux_amd64/aptly repo create foo

Local repo [foo] successfully added.
You can run 'aptly repo add foo ...' to add packages to repository.
$ ls -l /home/me/aptly
total 0
drwxr-xr-x 1 me me 78 Jul 21 08:43 db
```

The `db` directory has permission 0755. Since my umask is 0002, it should have 0775.

## Context
If Aptly obeyed its umask, I could use umask 0002 and an SGID directory to allow all users in a group to collaborate on maintaining a single repository.

## Possible Implementation
Pass 0777 to the directory creation syscall and 0666 to the file creation syscall, like most software does, rather than, I assume, 0755 and 0644.

## Your Environment
AMD64 Ubuntu Jammy, both using the packaged 1.4.0 and also the 1.5.0 from the GitHub release.

**Comment by Hawk777** [comment]
Yes, sorry, perhaps I worded it poorly; I’m quite well versed with how umask works, so I know that, from a technical perspective, a program is perfectly within its rights to choose whatever permission bits it wants. It’s just that, as a *convention*, programs usually use 0666 or 0777 unless there’s a good reason to do otherwise. Text editor? 0666. `touch`? 0666. Shell redirection? 0666. So what’s aptly’s reason for doing otherwise? Why does it behave differently from the majority of software, which lets the user have full control via umask?

**Comment by Hawk777** [comment]
As for using `sudo`, that seems like overkill (not to mention un-ergonomic, since you have to prepend every command with it). Groups are a thing, the very reason for them to exist is to allow multiple users to access shared resources. Why isn’t that the most obvious tool for this purpose?

**Comment by neolynx** [comment]
Hi and thanks for reporting !

I think this has already been changed back in 2016 to `0777`:
```
files/public.go:        return os.MkdirAll(filepath.Join(storage.rootPath, path), 0777)
files/public.go:        err := os.MkdirAll(poolPath, 0777)
```

at least I could not see any 775 mkdir except in test files. 

could you try with newly released 1.6.0 (https://github.com/aptly-dev/aptly/discussions/1414) ?

**Comment by neolynx** [comment]
closing as aptly cannot fix this
